client/client.go

```diff
@@ -1017,16 +1045,30 @@ func (c *Client) setupDrivers() error {
 		if err != nil {
 			return err
 		}
-		c.configLock.Lock()
-		applies, err := d.Fingerprint(c.config, c.config.Node)
-		c.configLock.Unlock()
+
+		request := &cstructs.FingerprintRequest{Config: c.config, Node: c.config.Node}
+		response := &cstructs.FingerprintResponse{
+			Attributes: make(map[string]string, 0),
+			Links:      make(map[string]string, 0),
+			Resources:  &structs.Resources{},
+		}
+
+		err = d.Fingerprint(request, response)
 		if err != nil {
 			return err
 		}
-		if applies {
+
+		// remove attributes we are supposed to skip
+		for _, attr := range skipped {
```

Comment:
If we don't care about thiese skipped attributes and are going to delete them from the response as soon as the fingerprint call is done, I would suggest pushing that logic one level down. You could have the FingerprintRequest also take a list of skipped(better name would be ignore) and have the implementors of `FingerPrint` ignore or remove anything in skipped before it returns. 

The current approach works, but it is not well encapsulated - its doing something, then mutating the response before using it.